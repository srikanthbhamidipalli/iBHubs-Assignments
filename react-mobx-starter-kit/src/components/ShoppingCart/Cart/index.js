import React, { Component } from "react";
import { observable } from "mobx";
import { observer } from "mobx-react";

import SelectedProduct from "./SelectedProduct";
import {
  SideNavBar,
  OutsideCartDiv,
  OutsideDivWithIcon,
  OutsideCartIcon,
  OutsideCartItemsCount,
  CrossIcon,
  SideNavBarBody,
  InsideCartIconDiv,
  InsideCartIcon,
  InsideCartItemsCount,
  CartItemsContainer,
  SubTotalContainer,
  AddToCartText,
  CheckOut
} from "./styledComponents";

@observer
class Cart extends Component {
  @observable isClicked = false;

  handleClick = () => {
    this.isClicked = !this.isClicked;
  };
  render() {
    let selectedItems = [];
    this.props.appStore.cartItemList.forEach(cartItem => {
      this.props.appStore.productList.forEach(product => {
        if (cartItem.productId === product.id)
          selectedItems.push(
            <SelectedProduct
              deleteItem={this.props.appStore.deleteCartItem}
              id={cartItem.productId}
              image={product.image}
              title={product.title}
              size={cartItem.size}
              style={product.style}
              price={product.price}
              quantity={cartItem.quantity}
            />
          );
      });
    });

    const outSideIcon = (
      <OutsideDivWithIcon>
        <OutsideCartIcon src="/assets/images/cart.jpg" alt="cartimage" />
        <OutsideCartItemsCount>
          <span>{this.props.appStore.cartListLength}</span>
        </OutsideCartItemsCount>
      </OutsideDivWithIcon>
    );

    return (
      <SideNavBar clickedStatus={this.isClicked}>
        <OutsideCartDiv onClick={this.handleClick}>
          {this.isClicked ? (
            <CrossIcon>
              <span>X</span>
            </CrossIcon>
          ) : (
            outSideIcon
          )}
        </OutsideCartDiv>
        <SideNavBarBody>
          <InsideCartIconDiv>
            <InsideCartIcon src="/assets/images/cart.jpg" alt="cartimage" />
            <span>Cart</span>
            <InsideCartItemsCount>
              <span>{this.props.appStore.cartListLength}</span>
            </InsideCartItemsCount>
          </InsideCartIconDiv>
          <CartItemsContainer>
            {selectedItems.length > 0 ? selectedItems : ""}
          </CartItemsContainer>
          <SubTotalContainer>
            <span>SUBTOTAL</span>
            <span>$ {this.props.appStore.subTotalAmount.toFixed(2)}</span>
          </SubTotalContainer>
          <CheckOut>
            <AddToCartText>CheckOut</AddToCartText>
          </CheckOut>
        </SideNavBarBody>
      </SideNavBar>
    );
  }
}

export default Cart;
