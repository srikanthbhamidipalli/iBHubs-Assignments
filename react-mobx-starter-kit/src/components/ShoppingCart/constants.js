export const sizesList = ["XS", "S", "M", "ML", "L", "XL", "XXL"];

export const orderByOptions = [
  "select",
  "Highest-To-Lowest",
  "Lowest-To-Highest"
];
