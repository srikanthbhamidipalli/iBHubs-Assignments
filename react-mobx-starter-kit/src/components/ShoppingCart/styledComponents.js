import styled from "styled-components";

export const HomeDivision = styled.div`
  display: flex;
  flex-direction: row;
`;

export const Division = styled.div``;
