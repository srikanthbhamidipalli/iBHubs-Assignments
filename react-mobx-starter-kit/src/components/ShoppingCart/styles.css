.size-header-product-display {
  display: flex;
  flex-direction: row;
}
