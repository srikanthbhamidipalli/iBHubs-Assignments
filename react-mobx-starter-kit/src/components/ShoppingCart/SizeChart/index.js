import React, { Component } from "react";
import { observer } from "mobx-react";

import { sizesChart } from "../constants";
import {
  SizesChart,
  SizesAndHeadingText,
  EachSizeItem,
  Division
} from "./styledComponents";

@observer
class SizeChart extends Component {
  handleClick = userselectedSizes => {
    this.props.appStore.addselectedSizes(userselectedSizes);
  };

  render() {
    let sizesList = [];
    Object.values(sizesChart).forEach(eachSize =>
      sizesList.push(
        <EachSizeItem
          key={eachSize}
          value={eachSize}
          isThisselectedSizes={this.props.appStore.selectedSizes.includes(
            eachSize
          )}
          onClick={() => this.handleClick(eachSize)}
        >
          {eachSize}
        </EachSizeItem>
      )
    );
    return (
      <SizesChart>
        <Division>
          <SizesAndHeadingText>Sizes:</SizesAndHeadingText>
        </Division>
        <Division>{sizesList}</Division>
      </SizesChart>
    );
  }
}

export default SizeChart;
