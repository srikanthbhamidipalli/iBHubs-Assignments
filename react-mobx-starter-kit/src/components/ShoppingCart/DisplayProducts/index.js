import React, { Component } from "react";
import { observer } from "mobx-react";

import EachProduct from "./EachProduct";
import { AllProductsContainer, PageStatus } from "./styledComponents";
import {
  fetchingProductsStatus,
  fetchingFailureMessage
} from "../../HybridUserForm/constants";

@observer
class DisplayProducts extends Component {
  render() {
    if (this.props.appStore.failureMessage !== fetchingFailureMessage) {
      return (
        <PageStatus className="page-status">
          <h1>Something Went Wrong!!!!</h1>
        </PageStatus>
      );
    } else if (
      this.props.appStore.productsFetchingStatus === fetchingProductsStatus
    ) {
      return (
        <PageStatus className="page-status">
          <h1>loading....</h1>
        </PageStatus>
      );
    } else if (
      this.props.appStore.productList.length === 0 &&
      this.props.appStore.failureMessage === fetchingFailureMessage
    ) {
      return (
        <PageStatus className="page-status">
          <h1>No products found!!!</h1>
        </PageStatus>
      );
    } else {
      let products = [];
      let productsList = this.props.appStore.orderByFilteredProducts;
      for (let i = 0; i < productsList.length; i++)
        products.push(
          <EachProduct
            key={productsList[i].id}
            appStore={this.props.appStore}
            eachItem={productsList[i]}
            addProductToCart={this.props.appStore.addProductToCart}
          />
        );
      return <AllProductsContainer>{products}</AllProductsContainer>;
    }
  }
}

export default DisplayProducts;
