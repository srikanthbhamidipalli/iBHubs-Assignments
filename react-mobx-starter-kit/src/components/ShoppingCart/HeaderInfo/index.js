import React, { Component } from "react";
import { observer } from "mobx-react";

import { orderByOptions } from "../constants";
import { SelectList, Option, HeaderContent, Span } from "./styledComponents";

@observer
class HeaderInfo extends Component {
  handleClick = e => {
    this.props.appStore.setOrderByValue(e.target.value);
  };
  render() {
    let optionsList = [];
    Object.values(orderByOptions).forEach(eachOption =>
      optionsList.push(
        <Option key={eachOption} value={eachOption}>
          {eachOption}
        </Option>
      )
    );
    const orderByOptionsList = (
      <SelectList onClick={this.handleClick}>{optionsList}</SelectList>
    );
    return (
      <HeaderContent>
        <Span id="products-count">
          {this.props.appStore.orderByFilteredProducts.length} Product(s) found
        </Span>
        <Span>Order By {orderByOptionsList}</Span>
      </HeaderContent>
    );
  }
}

export default HeaderInfo;
