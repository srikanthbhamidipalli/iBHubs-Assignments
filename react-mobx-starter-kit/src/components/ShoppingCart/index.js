import React from "react";
import { observer } from "mobx-react";

import SizeChart from "./SizeChart";
import HeaderInfo from "./HeaderInfo";
import DisplayProducts from "./DisplayProducts";
import Cart from "./Cart";
import { HomeDivision, Division } from "./styledComponents";
import { sizesChart } from "./constants";

@observer
class ShoppingCart extends React.Component {
  componentDidMount() {
    this.props.appStore.fetchProductsData();
  }
  render() {
    return (
      <HomeDivision>
        <Division>
          <SizeChart appStore={this.props.appStore} sizesChart={sizesChart} />
        </Division>
        <Division>
          <HeaderInfo appStore={this.props.appStore} />
          <DisplayProducts appStore={this.props.appStore} />
        </Division>
        <Cart appStore={this.props.appStore} />
      </HomeDivision>
    );
  }
}

export default ShoppingCart;
