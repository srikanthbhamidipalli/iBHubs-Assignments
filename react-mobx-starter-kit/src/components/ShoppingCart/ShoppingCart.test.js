import React from "react";
import { shallow } from "enzyme";
import ShoppingCart from "./index";
import appStore from "../../AppStore";
describe("ShoppingCart", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ShoppingCart debug appStore={appStore} />);

    expect(component).toMatchSnapshot();
  });
});
