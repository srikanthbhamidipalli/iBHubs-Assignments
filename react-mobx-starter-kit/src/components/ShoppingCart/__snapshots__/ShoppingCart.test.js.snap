// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart should render correctly in "debug" mode 1`] = `
<styled.div>
  <styled.div>
    <SizeChart
      appStore={
        AppStore {
          "accessToken": "",
          "cartItemList": Array [],
          "error": "",
          "failureMessage": "",
          "fetchingOptionsForLoginAndSignUp": [Function],
          "initialProductsFetchingStatus": "loading",
          "orderBy": "select",
          "productList": Array [],
          "selectedSizes": Array [],
        }
      }
    />
  </styled.div>
  <styled.div>
    <HeaderInfo
      appStore={
        AppStore {
          "accessToken": "",
          "cartItemList": Array [],
          "error": "",
          "failureMessage": "",
          "fetchingOptionsForLoginAndSignUp": [Function],
          "initialProductsFetchingStatus": "loading",
          "orderBy": "select",
          "productList": Array [],
          "selectedSizes": Array [],
        }
      }
    />
    <DisplayProducts
      appStore={
        AppStore {
          "accessToken": "",
          "cartItemList": Array [],
          "error": "",
          "failureMessage": "",
          "fetchingOptionsForLoginAndSignUp": [Function],
          "initialProductsFetchingStatus": "loading",
          "orderBy": "select",
          "productList": Array [],
          "selectedSizes": Array [],
        }
      }
    />
  </styled.div>
  <Cart
    appStore={
      AppStore {
        "accessToken": "",
        "cartItemList": Array [],
        "error": "",
        "failureMessage": "",
        "fetchingOptionsForLoginAndSignUp": [Function],
        "initialProductsFetchingStatus": "loading",
        "orderBy": "select",
        "productList": Array [],
        "selectedSizes": Array [],
      }
    }
  />
</styled.div>
`;
